lib/gems/pending/util/extensions/miq-benchmark.rb

```diff
@@ -22,15 +22,27 @@ def self.realtime_store(hash, key)
   # call completes.  If the hash already has a value for that key, the time is
   # accumulated.
   def self.realtime_block(key, &block)
-    outermost = !self.in_realtime_block?
     hash = current_realtime
-    self.current_realtime = hash if outermost
 
-    begin
+    if in_realtime_block?
       ret = realtime_store(hash, key, &block)
       return ret, hash
-    ensure
-      delete_current_realtime if outermost
+    else
+      # Outermost block.
+      begin
+        self.current_realtime = hash
+        begin
+          ret = realtime_store(hash, key, &block)
+          return ret, hash
+        ensure
+          delete_current_realtime
+        end
+      ensure
+        # A second layer of protection in case TimeoutError struck right after
+        # setting self.current_realtime, or right before `delete_current_realtime`.
+        # In those cases, current_realtime might (wrongly) still exist.
+        delete_current_realtime && in_realtime_block?
```

Comment:
@cben I don't understand what the `&& in_realtime_block?` is doing.

Is that needed?

Reply:
Was it intended to be `if in_realtime_block?` ?

Reply:
Oops bad rewrite, yes, s/&&/if/.
This means it's safe to do without checking :-)
Huh.  TIL ruby Hash#delete just returns nil if no such key.  I should have guessed but I keep getting surprised...

IMHO it's more readable with the check.

Reply:
Ok, cool.  I'm fine with keeping the `if in_realtime_block?` check.